deltas/tokenizers/tokenizer.py

```diff
@@ -58,3 +75,63 @@ def _tokenize(self, text, token_class=None):
                 tokens[value] = token
 
             yield token
+
+
+class TokenProcessor:
+    def segment(self, tokenized_text):
+        raise NotImplementedError()
+
+
+class CJKSegmenter(TokenProcessor):
+    """
+    Uses a cjk_lexicon to decide which tokenizer should be used
+    (Chinese, Japanese or Korean).
+    """
+    def __init__(self, cjk_lexicon):
+        self.regex_cjk = re.compile(cjk_lexicon['cjk'])
+        self.regex_japanese = re.compile(cjk_lexicon['japanese'])
+        self.regex_korean = re.compile(cjk_lexicon['korean'])
+
+    def segment(self, tokenized_text, token_class=None):
+        token_class = token_class or Token
+        text = "".join([tok[:] for tok in tokenized_text if tok.type == 'cjk_word']) # noqa
+        cjk_symbols = len(self.regex_cjk.findall(text))
+        jap_symbols = len(self.regex_japanese.findall(text))
+        kor_symbols = len(self.regex_korean.findall(text))
+        char_lang_frac = {'japanese': jap_symbols/cjk_symbols,
+                          'korean': kor_symbols/cjk_symbols}
+        max_char_lang_frac = max(char_lang_frac, key=char_lang_frac.get)
+        # check if at least 1/4 of chars are other than chinese,
+        # if not -> run chinese tokenizer
+        if char_lang_frac[max_char_lang_frac] > 0.25:
+            segmented_tokens = self._cjk_segmentation(tokenized_text, language=max_char_lang_frac, token_class=token_class) # noqa
+        else:
+            segmented_tokens = self._cjk_segmentation(tokenized_text, language='cjk', token_class=token_class) # noqa
+        return segmented_tokens
+
+    def _cjk_segmentation(self, tokenized_text, language, token_class=None):
+        token_class = token_class or Token
+        cjk_word_indices = list(filter(lambda x: tokenized_text[x].type == 'cjk_word', range(len(tokenized_text)))) # noqa
+
+        if language == 'cjk':
+            seg = ch_jieba.initialize()
```

Comment:
CH_JIEBA = None

```
def get_ch_jieba():
  if CH_JIEBA is None:
    CH_JIEBA = ch_jieba.initialize()
  return CH_JIEBA 

....

  seg = get_ch_jieba()
```

Reply:
jieba is a bit strange - the whole code(Tokenizer obj) is in jieba/__init__.py ; It took me more time than I expected...
FIXED for all tokenizers jieba/sudachi/konlpy